GeneralsMD/Code/Tools/WW3D/max2w3d/CMakeLists.txt

```diff
@@ -0,0 +1,190 @@
+    "ExportAllDlg.h"
+    "exportlog.cpp"
+    "exportlog.h"
+    "floaterdialog.cpp"
+    "floaterdialog.h"
+    "FormClass.cpp"
+    "FormClass.h"
+    "gamemaps.cpp"
+    "gamemaps.h"
+    "GameMtl.cpp"
+    "gamemtl.h"
+    "GameMtlDlg.cpp"
+    "GameMtlDlg.h"
+    "GameMtlForm.cpp"
+    "GameMtlForm.h"
+    "GameMtlPassDlg.cpp"
+    "GameMtlPassDlg.h"
+    "GameMtlShaderDlg.cpp"
+    "GameMtlShaderDlg.h"
+    "GameMtlTextureDlg.cpp"
+    "GameMtlTextureDlg.h"
+    "GameMtlVertexMaterialDlg.cpp"
+    "GameMtlVertexMaterialDlg.h"
+    "genlodextensiondialog.cpp"
+    "genlodextensiondialog.h"
+    "genmtlnamesdialog.cpp"
+    "genmtlnamesdialog.h"
+    "gennamesdialog.cpp"
+    "gennamesdialog.h"
+    "geometryexportcontext.h"
+    "geometryexporttask.cpp"
+    "geometryexporttask.h"
+    #"gmtldlg.cpp"
+    "gridsnapmodifier.cpp"
+    "gridsnapmodifier.h"
+    "hiersave.cpp"
+    "hiersave.h"
+    "hlodsave.cpp"
+    "hlodsave.h"
+    "InputDlg.cpp"
+    "InputDlg.h"
+    #"LightGlareSave.cpp"
+    #"LightGlareSave.h"
+    "logdlg.cpp"
+    "logdlg.h"
+    "max2w3d.def"
+    "maxworldinfo.cpp"
+    "maxworldinfo.h"
+    "meshbuild.cpp"
+    "meshbuild.h"
+    "meshcon.cpp"
+    "meshcon.h"
+    "MeshDeform.cpp"
+    "MeshDeform.h"
+    "MeshDeformData.cpp"
+    "MeshDeformData.h"
+    "MeshDeformDefs.h"
+    "MeshDeformPanel.cpp"
+    "MeshDeformPanel.h"
+    "MeshDeformSave.cpp"
+    "MeshDeformSave.h"
+    "MeshDeformSaveDefs.h"
+    "MeshDeformSaveSet.cpp"
+    "MeshDeformSaveSet.h"
+    "MeshDeformSet.cpp"
+    "MeshDeformSet.h"
+    "MeshDeformUndo.cpp"
+    "MeshDeformUndo.h"
+    "meshsave.cpp"
+    "meshsave.h"
+    "motion.cpp"
+    "motion.h"
+    "namedsel.cpp"
+    "namedsel.h"
+    "nullsave.cpp"
+    "nullsave.h"
+    "PCToPS2Material.cpp"
+    "presetexportoptionsdialog.cpp"
+    "presetexportoptionsdialog.h"
+    "PS2GameMtl.cpp"
+    "PS2GameMtlShaderDlg.cpp"
+    "PS2GameMtlShaderDlg.h"
+    "rcmenu.cpp"
+    "rcmenu.h"
+    "resource.h"
+    "SceneSetup.cpp"
+    "SceneSetupDlg.cpp"
+    "SceneSetupDlg.h"
+    "simpdib.cpp"
+    "simpdib.h"
+    "skin.cpp"
+    "skin.h"
+    "SkinCopy.cpp"
+    "skindata.cpp"
+    "skindata.h"
+    "SnapPoints.cpp"
+    "SnapPoints.h"
+    "TARGA.CPP"
+    "TARGA.H"
+    "util.cpp"
+    "util.h"
+    "Utility.cpp"
+    "vchannel.cpp"
+    "vchannel.h"
+    "vxl.cpp"
+    "vxl.h"
+    "vxldbg.cpp"
+    "vxldbg.h"
+    "vxllayer.cpp"
+    "vxllayer.h"
+    "w3dappdata.cpp"
+    "w3dappdata.h"
+    "w3ddesc.cpp"
+    "w3ddesc.h"
+    "w3ddlg.cpp"
+    "w3ddlg.h"
+    "w3dexp.cpp"
+    "w3dexp.h"
+    "w3dmtl.cpp"
+    "w3dmtl.h"
+    "w3dutil.cpp"
+    "w3dutil.h"
+    "w3d_file.h"
+    "w3d_obsolete.h"
+)
+
+add_library(max2w3d SHARED)
+set_target_properties(max2w3d PROPERTIES SUFFIX ".dle")
+
+target_include_directories(max2w3d PRIVATE
+    "../pluglib"
+    "../../../../../Dependencies/.maxsdk/Include" # Hack
+    "../../../../../Dependencies/.maxsdk/Include/Maxscrpt" # Hack
```

Comment:
max2w3d and pluglib currently contain these hacks. Maybe there is a better solution.

Reply:
Hmm, maybe add a cmake to the dependencies folder that creates a target to link against, but only if it finds the sdk files in the directory with it?